### PR TITLE
Add support for passing intl and other context variables to custom handlebar helpers

### PR DESCRIPTION
**Pull request by rikukissa**
Example usage in country config would now be 

```ts
import * as Handlebars from 'handlebars'
import { type IntlShape } from 'react-intl'

type HelperProps = { intl: IntlShape }

export function myIntlHelper({ intl }: HelperProps): Handlebars.HelperDelegate {
  return function (this: any, args: string[]) {
    const idParts = args.slice(0, -1)
    if (idParts.some((part) => part === undefined)) {
      return ''
    }

    const id = idParts.join('.')

    return intl.formatMessage({
      id,
      defaultMessage: 'Missing translation for ' + id
    })
  }
}
```

### CODE REVIEW

File: packages/client/src/utils/referenceApi.ts [hunk 1]
```diff
@@ -203,17 +204,25 @@ export async function importConditionals(): Promise<LoadConditionalsResponse> {
...
-  return handlebars
+async function importHandlebarHelpers(): Promise<LoadHandlebarHelpersResponse> {
+  try {
+    // https://github.com/rollup/plugins/tree/master/packages/dynamic-import-vars#limitations
+    const handlebars = await import(
+      /* @vite-ignore */ `${window.config.COUNTRY_CONFIG_URL}/handlebars.js`
+    )
+    return handlebars
+  } catch (error) {
+    return {}
```
Comment (rikukissa) [inline packages/client/src/utils/referenceApi.ts:224]: Definitely a big anti-pattern here. No sense to propagate the error upwards until we have a more detailed error than what `import` currently gives us

Comment (naftis) [reply packages/client/src/utils/referenceApi.ts:224]: Yeah would love to find a solution for error handling the imports :/ But should we throw an error anyway? Isn't the risk here that the certificates get unusable and there's not a visible reason for anyone why that's so

File: packages/client/src/views/PrintCertificate/PDFUtils.ts [hunk 2]
```diff
@@ -98,7 +98,20 @@ export function executeHandlebarsTemplate(
...
+  const customHelpers = getHandlebarHelpers()
+
+  for (const helperName of Object.keys(customHelpers)) {
+    /*
+     * Note for anyone adding new context variables to handlebar helpers:
+     * Everything you expose to country config's here will become API surface area,
+     * This means that countries will become dependant on it and it will be hard to remove or rename later on.
+     * If you need to expose the full record, please consider only exposing the specific values you know are needed.
+     * Otherwise what happens is that we lose the ability to refactor and remove things later on.
+     */
```
Comment (naftis) [inline packages/client/src/views/PrintCertificate/PDFUtils.ts:111]: Should this warning rather be in Farajaland, or are the context variables added to this file?

**Review by Zangetsu101 — APPROVED** [review]

File: packages/client/src/views/PrintCertificate/PDFUtils.ts (hunk 2 above)
Comment (rikukissa) [reply packages/client/src/views/PrintCertificate/PDFUtils.ts:111]: @naftis they are exposed from here, yes 👍

File: packages/client/src/utils/referenceApi.ts (hunk 1 above)
Comment (rikukissa) [reply packages/client/src/utils/referenceApi.ts:224]: Good point. That's a better default.

**Pull request merged**